python/interpret_community/dataset/dataset_wrapper.py

```diff
@@ -218,6 +238,21 @@ def summary_dataset(self):
         """
         return self._summary_dataset
 
+    def _set_default_index_cols(self, dataset):
+        if dataset.index.name is not None:
+            self._default_index_cols = dataset.index.name
+        if dataset.index.names is not None:
```

Comment:
Can they have both or can this be elif?

Reply:
apparently both
```
python
>>> from sklearn import datasets
>>> x, y = datasets.load_diabetes(return_X_y=True)
>>> nrows, ncols = x.shape
>>> column_names = [str(i) for i in range(ncols)]
>>> X = pd.DataFrame(x, columns=column_names)
>>> import uuid
>>> time_column_name = "Date" + str(uuid.uuid4())
>>> dates = pd.date_range('1980-01-01', periods=nrows, freq='MS')
>>> X[time_column_name] = dates
>>> index_keys = [time_column_name]
>>> X.set_index(index_keys, inplace=True)
>>> X
                                                 0         1         2         3         4         5         6         7         8         9
Date545baefd-7b22-4a53-9815-2e7ff4730f75
1980-01-01                                0.038076  0.050680  0.061696  0.021872 -0.044223 -0.034821 -0.043401 -0.002592  0.019908 -0.017646
1980-02-01                               -0.001882 -0.044642 -0.051474 -0.026328 -0.008449 -0.019163  0.074412 -0.039493 -0.068330 -0.092204
1980-03-01                                0.085299  0.050680  0.044451 -0.005671 -0.045599 -0.034194 -0.032356 -0.002592  0.002864 -0.025930
1980-04-01                               -0.089063 -0.044642 -0.011595 -0.036656  0.012191  0.024991 -0.036038  0.034309  0.022692 -0.009362
1980-05-01                                0.005383 -0.044642 -0.036385  0.021872  0.003935  0.015596  0.008142 -0.002592 -0.031991 -0.046641
...                                            ...       ...       ...       ...       ...       ...       ...       ...       ...       ...
2016-06-01                                0.041708  0.050680  0.019662  0.059744 -0.005697 -0.002566 -0.028674 -0.002592  0.031193  0.007207
2016-07-01                               -0.005515  0.050680 -0.015906 -0.067642  0.049341  0.079165 -0.028674  0.034309 -0.018118  0.044485
2016-08-01                                0.041708  0.050680 -0.015906  0.017282 -0.037344 -0.013840 -0.024993 -0.011080 -0.046879  0.015491
2016-09-01                               -0.045472 -0.044642  0.039062  0.001215  0.016318  0.015283 -0.028674  0.026560  0.044528 -0.025930
2016-10-01                               -0.045472 -0.044642 -0.073030 -0.081414  0.083740  0.027809  0.173816 -0.039493 -0.004220  0.003064

[442 rows x 10 columns]
>>> X.index
DatetimeIndex(['1980-01-01', '1980-02-01', '1980-03-01', '1980-04-01',
               '1980-05-01', '1980-06-01', '1980-07-01', '1980-08-01',
               '1980-09-01', '1980-10-01',
               ...
               '2016-01-01', '2016-02-01', '2016-03-01', '2016-04-01',
               '2016-05-01', '2016-06-01', '2016-07-01', '2016-08-01',
               '2016-09-01', '2016-10-01'],
              dtype='datetime64[ns]', name='Date545baefd-7b22-4a53-9815-2e7ff4730f75', length=442, freq=None)
>>> X.index.name
'Date545baefd-7b22-4a53-9815-2e7ff4730f75'
>>> X.index.names
FrozenList(['Date545baefd-7b22-4a53-9815-2e7ff4730f75'])
```

Reply:
let me just get rid of name for now and use names